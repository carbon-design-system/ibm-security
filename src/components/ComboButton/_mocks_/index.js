/**
 * @file Combo button mocks.
 * @copyright IBM Security 2019
 */

import { ArrowRight20, FolderDetails20 } from '@carbon/icons-react';

const actions = [
  {
    label: 'Start a task',
    iconDescription: 'arrow right',
    renderIcon: ArrowRight20,
    onClick: () => {},
  },
  {
    label: 'Add to folder',
    renderIcon: FolderDetails20,
    onClick: () => {},
  },
];

export default actions;
