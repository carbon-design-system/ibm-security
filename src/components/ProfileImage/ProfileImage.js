/**
 * @file Profile image.
 * @copyright IBM Security 2019
 */

import classnames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

import { getComponentNamespace } from '../../globals/namespace';

export const namespace = getComponentNamespace('profile-image');

/**
 * Profile image component.
 * @param {Object.<string, *>} props Profile image props.
 * @returns {ProfileImage} Profile image instance.
 */
const ProfileImage = ({ className, large, profile, ...other }) => {
  const classNames = classnames(namespace, className, {
    [`${namespace}--large`]: large,
  });

  return profile.image_url ? (
    <img
      className={classNames}
      alt={`${profile.name.first_name} ${profile.name.surname}`}
      src={profile.image_url}
      {...other}
    />
  ) : (
    <span className={classNames} {...other}>
      {`${profile.name.first_name.charAt(0)}${profile.name.surname.charAt(0)}`}
    </span>
  );
};

ProfileImage.defaultProps = {
  className: null,
  large: false,
};

ProfileImage.propTypes = {
  /** @type {string} Extra classes. */
  className: PropTypes.string,

  /** @type {boolean} Determines whether the profile image is large or not. */
  large: PropTypes.bool,

  /** @type {Object.<string, *>} An object list of profile information. */
  profile: PropTypes.shape({
    /** @type {string} Profile image URL. */
    image_url: PropTypes.string,

    /** @type {Object.<string, string>} An object list of profile name information. */
    name: PropTypes.shape({
      /** @type {string} First name. */
      first_name: PropTypes.string.isRequired,

      /** @type {string} Surname. */
      surname: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};

export default ProfileImage;
