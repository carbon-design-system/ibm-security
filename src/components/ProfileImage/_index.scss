////
/// Profile image component.
/// @group profile-image
/// @copyright IBM Security 2019
////

@import '../Component/mixins';
@import 'mixins';

@include security--component($name: profile-image) {
  @include profile-image;
}
