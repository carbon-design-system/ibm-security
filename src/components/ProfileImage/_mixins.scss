////
/// Profile image mixins.
/// @group profile-image
/// @copyright IBM Security 2019
////

@import '@carbon/layout/scss/mini-unit';
@import '@carbon/themes/scss/tokens';
@import '@carbon/type/scss/styles';

@mixin profile-image {
  @include carbon--type-style($name: body-short-01);
  display: inline-flex;
  width: carbon--mini-units($count: 3.5);
  height: carbon--mini-units($count: 3.5);
  color: $inverse-01;
  align-items: center;
  background-color: $icon-02;
  border-radius: 50%;
  justify-content: center;
  object-fit: cover;
  text-transform: uppercase;

  &--large {
    @include carbon--type-style($name: productive-heading-03);
    width: carbon--mini-units($count: 7);
    height: carbon--mini-units($count: 7);
  }
}
