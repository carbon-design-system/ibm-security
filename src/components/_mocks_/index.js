/**
 * @file Global mocks.
 * @copyright IBM Security 2018
 */

import React from 'react';

import random from '../../globals/random';

const label = 'Label';

const lorem =
  'Lorem ipsum dolor sit, amet consectetur adipisicing elit. Eveniet rerum aliquid perferendis, nulla nam ad excepturi, iure earum dolorum ipsa quae cum voluptatibus nemo quis debitis, aperiam repudiandae dolore deleniti.';

const icon =
  'data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZpZXdCb3g9IjAgMCAzMiAzMiI+ICA8cGF0aCBmaWxsPSIjZmZmIiBkPSJNMTYgMThINmEyIDIgMCAwIDEtMi0yVjZhMiAyIDAgMCAxIDItMmgxMGEyIDIgMCAwIDEgMiAydjEwYTIgMiAwIDAgMS0yIDJ6TTYgNnYxMGgxMFY2em0yMCA2djRoLTR2LTRoNG0wLTJoLTRhMiAyIDAgMCAwLTIgMnY0YTIgMiAwIDAgMCAyIDJoNGEyIDIgMCAwIDAgMi0ydi00YTIgMiAwIDAgMC0yLTJ6bTAgMTJ2NGgtNHYtNGg0bTAtMmgtNGEyIDIgMCAwIDAtMiAydjRhMiAyIDAgMCAwIDIgMmg0YTIgMiAwIDAgMCAyLTJ2LTRhMiAyIDAgMCAwLTItMnptLTEwIDJ2NGgtNHYtNGg0bTAtMmgtNGEyIDIgMCAwIDAtMiAydjRhMiAyIDAgMCAwIDIgMmg0YTIgMiAwIDAgMCAyLTJ2LTRhMiAyIDAgMCAwLTItMnoiLz48L3N2Zz4=';

const id = '0';

const disabled = true;

const invalid = true;

const invalidText = 'Invalid text';

/**
 * Returns a child component.
 */
const children = <span>{label}</span>;

const className = 'class';

/**
 * Creates a URL for mock data.
 * @param {string} [hash=''] - A hash to pass to the URL method.
 * @returns {string} The mock data URL.
 */
const url = (hash = '') => `#${hash}`;

/**
 * Creates a `href` for mock data.
 * @returns {string} The `href` to return.
 */
const href = () => url();

export {
  children,
  className,
  disabled,
  href,
  icon,
  id,
  invalid,
  invalidText,
  label,
  lorem,
  random,
  url,
};
