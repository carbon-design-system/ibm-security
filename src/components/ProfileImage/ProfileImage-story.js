/**
 * @file Profile image stories.
 * @copyright IBM Security 2019 - 2021
 */

import { boolean, object } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import React from 'react';

import { security } from '../../../.storybook';

import { ProfileImage } from '../..';

import { className, large, profile } from './_mocks_';

const profileImageProps = () => ({
  className,
  large: boolean('Large (large)', large),
});

storiesOf(security('ProfileImage'), module).add('default', () => (
  <ProfileImage
    {...profileImageProps()}
    profile={object('Profile (profile)', profile)}
  />
));
