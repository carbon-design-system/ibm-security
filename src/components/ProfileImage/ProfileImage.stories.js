/**
 * @file Profile image stories.
 * @copyright IBM Security 2019
 */

import { checkA11y } from '@storybook/addon-a11y';
import centered from '@storybook/addon-centered/react';
import { boolean, object } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';

import React from 'react';

import { components } from '../../../.storybook';

import { ProfileImage } from '../..';

import { className, large, profile } from './_mocks_';

const profileImageProps = () => ({
  className,
  large: boolean('Large (large)', large),
});

storiesOf(components('ProfileImage'), module)
  .addDecorator(checkA11y)
  .addDecorator(centered)
  .add('default', () => (
    <ProfileImage
      {...profileImageProps()}
      profile={object('Profile (profile)', profile)}
    />
  ));
