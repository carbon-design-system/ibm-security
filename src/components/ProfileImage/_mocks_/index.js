/**
 * @file Profile image mocks.
 * @copyright IBM Security 2019
 */

import { className } from '../../_mocks_';
import { profile } from '../../Header/_mocks_';

const large = true;

export { className, large, profile };
