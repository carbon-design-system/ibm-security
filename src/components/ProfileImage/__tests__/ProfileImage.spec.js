/**
 * @file Profile image tests.
 * @copyright IBM Security 2019
 */

import { shallow } from 'enzyme';
import React from 'react';

import { className, large, profile } from '../_mocks_';

import ProfileImage from '..';

describe('ProfileImage', () => {
  let profileImage;

  beforeEach(() => {
    profileImage = shallow(
      <ProfileImage className={className} profile={profile} />
    );
  });

  it('renders', () => {
    expect(profileImage).toMatchSnapshot();
  });

  it('renders the large variation', () => {
    profileImage.setProps({ large });
    expect(profileImage.render()).toMatchSnapshot();
  });
});
