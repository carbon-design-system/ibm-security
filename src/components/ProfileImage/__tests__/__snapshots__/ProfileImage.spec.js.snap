// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileImage renders 1`] = `
<span
  className="security--profile-image class"
>
  SU
</span>
`;

exports[`ProfileImage renders the large variation 1`] = `
<span
  class="security--profile-image class security--profile-image--large"
>
  SU
</span>
`;
